core/server/apps/amp/lib/router.js

```diff
@@ -42,21 +43,42 @@ function getPostData(req, res, next) {
             next();
         })
         .catch(function (err) {
-            if (err instanceof errors.NotFoundError) {
-                return next(err);
-            }
-
             next(err);
         });
 }
 
+function checkIfAMPIsEnabled(req, res, next) {
+    var ampIsEnabled = config.theme.amp;
+
+    if (ampIsEnabled) {
+        return next();
+    }
+
+    // CASE: we don't support amp pages for static pages
+    if (req.body.post && req.body.post.page) {
+        return errors.error404(req, res, next);
+    }
+
+    /**
+     * CASE: amp is disabled, we redirect the user to the requested post
```

Comment:
That's exactly what I was wondering as well: will it confuse the search engines, if we send a 404? What if users enable it again? Will google e. g. know, that the AMP page *does* exist again, if we sent 404 beforehand for the same page requests?